docs/pipeline-configuration-files/specifications.mdx

```diff
@@ -0,0 +1,272 @@
+- **Default**: None
+- **Description**: The node contains a list of [`processor`](#processor) definitions. These processors process all
+  records flowing through the pipeline. They are executed after source processors and before destination processors. The
+  processors are executed in the order in which they are defined in the list.
+
+## connector
+
+- **Data Type**: Mapping node
+- **Required**: n/a
+- **Default**: None
+- **Description**: This node contains the configuration of a single connector.
+
+### id
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: ID of the [`connector`](#connector). It should uniquely identify the connector inside the pipeline.
+  If multiple connectors inside the pipeline use the same ID, provisioning will fail.
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### type
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Allowed Values**: `source`, `destination`
+- **Description**: Defines if the connector is a source or a destination. A working pipeline needs at least one source
+  and one destination.
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### plugin
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: This node references the connector plugin. See how to [reference a connector](/docs/connectors/referencing).
+
+**Warning**: Changing this property will cause the connector to start from the beginning.
+
+### name
+
+- **Data Type**: String
+- **Required**: No
+- **Default**: Same as connector [`id`](#id-1)
+- **Description**: Human readable name for the connector.
+
+### settings
+
+- **Data Type**: Mapping node
+- **Required**: Yes
+- **Default**: None
+- **Description**: A map of configuration keys and values for the connector plugin. The values in this map depend on the
+  chosen connector [`type`](#type) and [`plugin`](#plugin-1).
+
+### processors
+
+- **Data Type**: Sequence node containing [`processor`](#processor) nodes
+- **Required**: No
+- **Default**: None
+- **Description**: The node contains a list of [`processor`](#processor) definitions. These processors only process
+  records coming from or flowing to a source or destination. The processors are executed in the order in which they are
+  defined in the list.
+
+## processor
+
+- **Data Type**: Mapping node
+- **Required**: n/a
+- **Default**: None
+- **Description**: This node contains the configuration of a single processor.
+
+### id
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: ID of the [`processor`](#processor). It should uniquely identify the processor inside the pipeline.
+  If multiple processors inside the pipeline use the same ID, provisioning will fail.
+
+### type
+
+- **Data Type**: String
+- **Required**: Yes
+- **Default**: None
+- **Description**: Defines the processor type.
```

Comment:
Would be good to also say the available types or where to find the available types.

Reply:
I added a link to the processor documentation that we have right now, but we don't have a full list of processors, we just point to the [godoc](https://pkg.go.dev/github.com/conduitio/conduit/pkg/processor/procbuiltin) there. I think we can and should do better by having a full list of processors and their capabilities, similar to the [kafka connect docs](https://docs.confluent.io/platform/current/connect/transforms/overview.html). That's actually what [this issue](https://github.com/ConduitIO/conduit/issues/966) was meant for.